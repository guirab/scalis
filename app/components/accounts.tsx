"use client";
import { AccountsContext } from "@/store/context";
import Link from "next/link";
import { useContext, useEffect, useState } from "react";

export default function Accounts() {
  const [showAccounts, setShowAccounts] = useState(false);
  const { accounts, setAccounts } = useContext(AccountsContext);

  useEffect(() => {
    if (!showAccounts) return;
    fetch("/api/accounts")
      .then((res) => res.json())
      .then((data) => setAccounts(data))
      .catch((error) => console.error("Error fetching accounts:", error));
  }, [showAccounts]);

  return (
    <div
      className="flex items-center flex-col w-fit mt-4 group group-[data-show=true] bg-white bg-opacity-30 rounded-lg h-fit p-4"
      data-show={showAccounts}
    >
      <span
        className="text-2xl cursor-pointer w-fit"
        onClick={() => setShowAccounts(!showAccounts)}
      >
        Accounts
      </span>
      <div className="flex flex-col w-full group-data-[show=false]:hidden">
        {accounts.map((account: any) => (
          <div
            key={account.id}
            className="flex w-full mt-4 gap-x-4 justify-between"
          >
            <span>Username: {account.username}</span>
            <Link href="/account/login">View Account</Link>
          </div>
        ))}
      </div>
    </div>
  );
}
