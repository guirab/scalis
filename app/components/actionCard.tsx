import { DepositView } from "./cardView/deposit";
import { WithdrawView } from "./cardView/withdraw";
import { TransferView } from "./cardView/transfer";

export const ActionCard: React.FC<React.HTMLProps<any> & ActionCardType> = ({
  action,
  open,
  setOpen,
  ...props
}) => {
  const currentView = {
    deposit: <DepositView open={open} setOpen={setOpen} />,
    withdraw: <WithdrawView open={open} setOpen={setOpen} />,
    transfer: <TransferView open={open} setOpen={setOpen} />,
  }[action as string];

  return (
    <>
      {open && (
        <div
          className="block w-full mt-4 space-y-4 max-w-[240px]"
          data-testid={`${action}-view`}
          {...props}
        >
          {currentView}
        </div>
      )}
    </>
  );
};
