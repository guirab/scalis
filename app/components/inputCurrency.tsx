export const InputCurrency: React.FC<
  React.HTMLProps<any> & InputCurrencyType
> = ({ value, setValue, ...props }) => {
  const currencyMask = (e: React.ChangeEvent<HTMLInputElement>) => {
    let v = e.target.value;
    v = v.replace(/\D/g, "");
    v = v.replace(/(\d)(\d{2})$/, "$1.$2");
    v = v.replace(/(?=(\d{3})+(\D))\B/g, ",");
    setValue(v);
  };
  return (
    <div className="relative w-full">
      <span className="absolute text-black left-px">$&nbsp;</span>
      <input
        autoComplete="off"
        type="text"
        onChange={(e) => currencyMask(e)}
        value={value}
        name={props.name}
        id={props.id}
        required
        className={`text-black pl-4 ${props.className} w-full`}
      />
    </div>
  );
};
