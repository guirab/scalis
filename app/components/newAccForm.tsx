"use client";
import { useState } from "react";
import { create } from "../actions";

export default function NewAccForm() {
  const [newAcc, setNewAcc] = useState(false);

  async function onClick(e: FormData) {
    await create(e).then(() => setNewAcc(false));
  }

  return (
    <div
      className="flex flex-col items-center w-full mt-4 group group-[data-newacc=true]"
      data-newacc={newAcc}
    >
      <span
        className="text-2xl cursor-pointer w-fit"
        onClick={() => setNewAcc(!newAcc)}
      >
        New Account
      </span>
      <form
        action={onClick}
        className="flex flex-col group-data-[newacc=false]:hidden"
      >
        <label htmlFor="username">Username</label>
        <input
          autoComplete="off"
          type="text"
          name="username"
          id="username"
          required
          className="text-black pl-2"
        />
        <label htmlFor="password">Password</label>
        <input
          autoComplete="off"
          type="password"
          name="password"
          id="password"
          required
          className="text-black pl-2"
        />
        <label htmlFor="checking">Checking Balance</label>
        <input
          autoComplete="off"
          type="number"
          name="checking"
          id="checking"
          required
          className="text-black pl-2"
        />
        <label htmlFor="savings">Savings Balance</label>
        <input
          autoComplete="off"
          type="number"
          name="savings"
          id="savings"
          required
          className="text-black pl-2"
        />
        <button type="submit" className="bg-green-600 mt-4">
          Create Account
        </button>
      </form>
    </div>
  );
}
