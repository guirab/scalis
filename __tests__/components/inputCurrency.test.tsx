import React from "react";
import { render, fireEvent, getByRole, cleanup } from "@testing-library/react";
import { InputCurrency } from "../../app/components/inputCurrency";
import { afterEach, describe, expect, it, vitest } from "vitest";

describe("InputCurrency", () => {
  afterEach(() => {
    cleanup();
  });

  it("snapshot", () => {
    const { container } = render(
      <InputCurrency
        value=""
        setValue={() => {}}
        name="currency"
        id="currency"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should update the value when input is changed", () => {
    const setValue = vitest.fn();
    const { getByTestId } = render(
      <InputCurrency
        value=""
        setValue={setValue}
        name="currency"
        id="currency"
      />
    );

    const input = getByTestId("currency-input");
    fireEvent.change(input, { target: { value: "12345" } });
    expect(setValue).toHaveBeenCalledWith("123.45");
  });
});
