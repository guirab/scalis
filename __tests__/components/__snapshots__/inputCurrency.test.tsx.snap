// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputCurrency > snapshot 1`] = `
<div>
  <div
    class="relative w-full"
  >
    <span
      class="absolute text-black left-px"
    >
      $Â 
    </span>
    <input
      autocomplete="off"
      class="text-black pl-4 undefined w-full"
      data-testid="currency-input"
      id="currency"
      name="currency"
      required=""
      type="text"
      value=""
    />
  </div>
</div>
`;
